cmd/etcd/list.go

```diff
@@ -31,10 +31,21 @@ func etcdListCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "member-list",
 		Short: "Returns etcd cluster members list",
+		PreRun: func(cmd *cobra.Command, args []string) {
+			// ensure logs don't mess up the output
+			logrus.SetOutput(cmd.ErrOrStderr())
+		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			c := config.GetCmdOpts()
-			ctx := context.Background()
-			etcdClient, err := etcd.NewClient(c.K0sVars.CertRootDir, c.K0sVars.EtcdCertDir, c.NodeConfig.Spec.Storage.Etcd)
+			opts, err := config.GetCmdOpts(cmd)
+			if err != nil {
+				return err
+			}
+			nodeConfig, err := opts.K0sVars.NodeConfig()
+			if err != nil {
+				return err
+			}
+			ctx := cmd.Context()
```

Comment:
Using the cmd.Context in all commands as the root context would allow us to add a global `--timeout` flag, the root persistent-prerun would do `ctx, cancel := context.WithTimeout(cmd.Ctx, time.Parse(&timeOut)); cmd.SetContext(ctx)` 🤔